jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt

```diff
@@ -23,7 +23,7 @@ class DeleteObjectAction(private val project: Project, treeTable: S3TreeTable) :
         deleteNodes(project, treeTable, nodes.filterIsInstance<S3TreeObjectNode>())
     }
 
-    override fun enabled(nodes: List<S3TreeNode>): Boolean = nodes.all { it is S3TreeObjectNode }
+    override fun enabled(nodes: List<S3TreeNode>): Boolean = nodes.all { it::class == S3TreeObjectNode::class }
```

Comment:
TODO: support deleting versioned objects?

Reply:
A added TODO, will be happy to support that in next iteration once this chunk of changes is all good to go :)